src/main/java/relay/data_access/FirebaseCourseDataAccessObject.java

```diff
@@ -0,0 +1,114 @@
+
+package relay.data_access;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+
+import relay.entity.Course;
+import relay.entity.Instructor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+
+public class FirebaseCourseDataAccessObject {
+
+
+    /**
+     * Creates a new course document in Firestore based on the given Course object.
+     *
+     * @param course The Course object containing course details to be added to Firestore.
+     */
+    public void createCourse(Course course) {
+        Map<String, Object> courseDocument = new HashMap<>();
+        courseDocument.put("courseID", course.getCourseID());
+        courseDocument.put("courseName", course.getCourseName());
+        courseDocument.put("instructorID", course.getInstructor().getInstructorID());
+
+        ApiFuture<WriteResult> future = FirestoreSingleton.get().collection("courses").document(course.getCourseID()).set(courseDocument);
```

Comment:
Change this to  `ApiFuture<WriteResult> future = FirestoreSingleton.get().collection("courses").addData(courseDocument);` in order to let firebase set the courseID.

Then, retrieve the newly generated courseID with `future.get().getId()` and save that into the `Course` object with `setCourseID` method.